tests/test_plan.py

```
@@ -0,0 +1,294 @@
+
+    def test_extract_json_markdown_code_block(self):
+        from core.utils import extract_json
+
+        text = '```json\n{"steps": [{"step": 1}], "total_steps": 1}\n```'
+        result = extract_json(text)
+        assert result["total_steps"] == 1
+
+    def test_extract_json_invalid(self):
+        from core.utils import extract_json
+
+        with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)